src/App.tsx

```diff
@@ -1,14 +1,43 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { getFilterTodos } from './utils/getFilterTodos';
+import { Status } from './types/Status';
+
+const todosFilter = (visibleTodos: Todo[], query: string) => {
+  return visibleTodos.filter(todo =>
+    todo.title.toLowerCase().includes(query.trim().toLowerCase()),
+  );
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [filter, setFilter] = useState<Status>(Status.All);
+  const [query, setQuery] = useState('');
+
+  let visibleTodos = getFilterTodos([...todos], filter);
+
+  if (query) {
+    visibleTodos = todosFilter(visibleTodos, query);
+  }
```

Comment:
I meant to remove filter logic from the component at all. And You don't need to do a copy manually as filter doesn't mutate an array